codeworld-api/src/CodeWorld/Picture.hs

```diff
@@ -213,16 +213,15 @@ rectangleVertices w h = [ (w / 2, h / 2), (w / 2, -h / 2), (-w / 2, -h / 2), (-w
 
 -- | A thin rectangle, with this width and height
 rectangle :: HasCallStack => Double -> Double -> Picture
-rectangle ps = rectangleToPolygon callStack ps
+rectangle w h = Rectangle callStack w h
```

Comment:
This is correct if you want to leave it this way.  But just as a comment, notice that you can eta reduce these to:

    rectangle = Rectangle callStack

which is the pattern Gabe was suggesting you follow.  It means the same thing as what you wrote, but you're defining the whole function at once by partially applying the constructor, instead of defining it in terms of its arguments.

Reply:
Ohhh..I see it now. Thank you!

Reply:
so it seems like lines 487 to 508 should mention width and height somewhere...for example, Rectangle mentions only "width" on line 491. But it type-checked.